packages/contracts/contracts/CollateralManager.sol

```diff
@@ -0,0 +1,372 @@
+            memory collateralInfos = getCollateralInfo(_bidId);
+        address borrower = tellerV2.getLoanBorrower(_bidId);
+        (validation_, ) = _checkBalances(borrower, collateralInfos, true);
+    }
+
+    function checkBalances(
+        address _borrowerAddress,
+        ICollateralEscrowV1.Collateral[] calldata _collateralInfo
+    ) public returns (bool validated_, bool[] memory checks_) {
+        return _checkBalances(_borrowerAddress, _collateralInfo, false);
+    }
+
+    /**
+     * @notice Deploys a new collateral escrow and deposits collateral.
+     * @param _bidId The associated bidId of the collateral escrow.
+     */
+    function deployAndDeposit(uint256 _bidId) external {
+        if (_isBidCollateralBacked[_bidId]) {
+            require(_msgSender() == address(tellerV2), 'Sender not authorized');
+            (address proxyAddress, ) = _deployEscrow(_bidId);
+            _escrows[_bidId] = proxyAddress;
+
+            for (
+                uint256 i;
+                i < _bidCollaterals[_bidId].collateralAddresses.length();
+                i++
+            ) {
+                _deposit(
+                    _bidId,
+                    _bidCollaterals[_bidId].collateralInfo[
+                        _bidCollaterals[_bidId].collateralAddresses.at(i)
+                    ]
+                );
+            }
+
+            emit CollateralEscrowDeployed(_bidId, proxyAddress);
+        }
+    }
+
+    /**
+     * @notice Gets the address of a deployed escrow.
+     * @notice _bidId The bidId to return the escrow for.
+     * @return The address of the escrow.
+     */
+    function getEscrow(uint256 _bidId) external view returns (address) {
+        return _escrows[_bidId];
+    }
+
+    /**
+     * @notice Gets the collateral info for a given bid id.
+     * @param _bidId The bidId to return the collateral info for.
+     * @return infos_ The stored collateral info.
+     */
+    function getCollateralInfo(uint256 _bidId)
+        public
+        view
+        returns (ICollateralEscrowV1.Collateral[] memory infos_)
+    {
+        CollateralInfo storage collateral = _bidCollaterals[_bidId];
+        address[] memory collateralAddresses = collateral
+            .collateralAddresses
+            .values();
+        infos_ = new ICollateralEscrowV1.Collateral[](
+            collateralAddresses.length
+        );
+        for (uint256 i; i < collateralAddresses.length; i++) {
+            infos_[i] = collateral.collateralInfo[collateralAddresses[i]];
+        }
+    }
+
+    function deposit(
+        uint256 _bidId,
+        ICollateralEscrowV1.Collateral[] calldata _collateral
+    ) external {
+        commitCollateral(_bidId, _collateral);
+    }
+
+    /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     */
+    function withdraw(uint256 _bidId) external {
+        if (_isBidCollateralBacked[_bidId]) {
```

Comment:
this can just be a require statement   not an if block

Reply:
same as `lenderAcceptBid`